calico_containers/pycalico/netns.py

```diff
@@ -105,13 +93,10 @@ def remove_ip_from_interface(container_pid, ip, interface_name,
     :return: None. raises CalledProcessError on error.
     """
     with NamedNamespace(container_pid, proc=proc_alias) as ns:
-        ns.check_call("ip -%(version)s addr del %(addr)s/%(len)s "
-                      "dev %(device)s" %
-                        {"version": ip.version,
-                         "len": PREFIX_LEN[ip.version],
-                         "addr": ip,
-                         "device": interface_name},
-                      shell=True)
+        ns.check_call(["ip", "-%s" % ip.version, "addr", "del",
+                       "%s/%s" % (ip, PREFIX_LEN[ip.version]),
+                       "dev", "%(device)s" % interface_name],
+                      timeout=IP_CMD_TIMEOUT)
 
 
 def set_up_endpoint(ip, hostname, orchestrator_id, workload_id, cpid, next_hop_ips,
```

Comment:
This method is wrong and it would be nice to clean it up.

It shouldn't be creating an Endpoint object - the caller should be doing that. The netns code shouldn't depend on the Endpoint class.